### PR TITLE
EPMLSTRCMW-162 Implement json formatter for algebraic data types

### PR DESCRIPTION
**Pull request by myazinn**

### CODE REVIEW

File: utils/src/main/scala/cromwell/pipeline/utils/json/AdtJsonFormatter.scala [hunk 1]
```diff
@@ -0,0 +1,95 @@
+package cromwell.pipeline.utils.json
+
+import cromwell.pipeline.utils.json.AdtJsonFormatter._
+import play.api.libs.json._
+
+import scala.reflect.{ClassTag, classTag}
```
Comment (GrigoriiBerezin) [inline utils/src/main/scala/cromwell/pipeline/utils/json/AdtJsonFormatter.scala:6]: Formatter should left space between brackets and letters

File: utils/src/test/scala/cromwell/pipeline/utils/json/AdtJsonFormatterTest.scala [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        val json = """["one", "two"]"""
+        an[Exception] should be thrownBy Json.parse(json).as[DummyTrait]
+      }
+
+      "instance does not contain typeFieldName key" in {
+        val json = """{"firstAField":"StringField","secondAField":1}"""
+        an[Exception] should be thrownBy Json.parse(json).as[DummyTrait]
+      }
+
+      "instance contain unexisting typeFieldName" in {
```
Comment (GrigoriiBerezin) [inline utils/src/test/scala/cromwell/pipeline/utils/json/AdtJsonFormatterTest.scala:82]: instance contain nonexistent typeFieldName

**Review by GrigoriiBerezin — COMMENTED** [review]
What a masterpiece! Left some comments

File: utils/src/main/scala/cromwell/pipeline/utils/json/AdtJsonFormatter.scala (hunk 1 above)
Comment (myazinn) [reply utils/src/main/scala/cromwell/pipeline/utils/json/AdtJsonFormatter.scala]: Ooops. Fixed.

File: utils/src/test/scala/cromwell/pipeline/utils/json/AdtJsonFormatterTest.scala (hunk 2 above)
Comment (myazinn) [reply utils/src/test/scala/cromwell/pipeline/utils/json/AdtJsonFormatterTest.scala]: It looks like both words are suitable in this case, but I like your option more than mine. Fixed.

**Review by GrigoriiBerezin — APPROVED** [review]
Thank you for that realisation

**Review by IaroslavTavchenkov — APPROVED** [review]
Nice work!
Thank you!)

**Review by Dmitrii-Shevchenko — APPROVED** [review]
Perfect work!

**Review by mnesvit — APPROVED** [review]

**Review by iliaonishchenko — APPROVED** [review]

**Pull request merged**